soroban-sdk/src/prng.rs

```diff
@@ -0,0 +1,48 @@
+//! Prng contains functions for pseudo-random functions.
+use crate::{
+    env::internal, unwrap::UnwrapInfallible, Bytes, Env, IntoVal, TryIntoVal, Val,
+    Vec,
+};
+
+/// Prng provides access to pseudo-random  functions.
+pub struct Prng {
+    env: Env,
+}
+
+impl Prng {
+    pub(crate) fn new(env: &Env) -> Prng {
+        Prng { env: env.clone() }
+    }
+
+    pub fn env(&self) -> &Env {
+        &self.env
+    }
+
+    // Reseeds the pseudorandom number generator (PRNG) with the provided `seed` value.
+    pub fn prng_reseed(&self, seed: &Bytes) {
```

Comment:
@graydon Can we name this function `seed` instead of `reseed`? Most APIs I've seen simply call it seed, because the action is seeding the generator. The fact it was already seeded by default, which the `re` signals, is a concern of the generator in general, and not a concern of this specific function.
```suggestion
    pub fn seed(&self, seed: &Bytes) {
```